### PR TITLE
Add a proof-of-concept TensorFlow example demonstrating the use of ML to predict material properties

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: examples/machine_learning/neural_network_train.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python
```
Comment (timurbazhirov) [inline examples/machine_learning/neural_network_train.py:1]: I suggest creating "other" directory to hold the sources and "assets" subdirectory inside it where the data would reside

File: examples/machine_learning/neural_network_train.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+random.seed(1234)
+
+
+# Next, we'll get everything into a Pandas dataframe.
+
+# In[]:
+
+
+# Load data into a Pandas dataframe for convenience
+df = pd.read_csv("Data.csv", sep=",")
```
Comment (timurbazhirov) [inline examples/machine_learning/neural_network_train.py:93]: Data should be previewed

File: examples/machine_learning/neural_network_train.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+# ## Hyperparameter Optimization
+# 
+# In this example, we use Scaled-Exponential Linear Unit activation functions, which [provably](https://arxiv.org/pdf/1706.02515.pdf) result in a self-normalizing effect as information flows through a neural network's layers, provided the network is sufficiently deep. One assumption made in the proof is the use of LeCunn's initialization strategy for weights and biases.
+
+# In[]:
+
+
+# Weight initialization strategy
+initializer = tf.keras.initializers.lecun_normal()
```
Comment (timurbazhirov) [inline examples/machine_learning/neural_network_train.py:109]: Per our discussion today - some of the details re choices for the initialization and/or activation deserve more explanation. Pls add.

File: examples/machine_learning/neural_network_train.py (hunk 1 above)
Comment (AcylSilane) [reply examples/machine_learning/neural_network_train.py:1]: Sounds like a good idea. What do you think about editing our post-save hook to drop the .py files in a centralized `other` (maybe we call this src) directory (preserving folder order)? So, the directory structure would end up looking like this:

--job
--machine_learning
--material
--system
--workflow
--assets
--other (or src)
||--job
||--machine_learning
||--material
||--system
||--workflow

Comment (AcylSilane) [reply examples/machine_learning/neural_network_train.py]: New directory structure is created in most recent commits

File: examples/machine_learning/neural_network_train.py (hunk 2 above)
Comment (AcylSilane) [reply examples/machine_learning/neural_network_train.py]: Data is now previewed

File: examples/machine_learning/neural_network_train.py (hunk 3 above)
Comment (AcylSilane) [reply examples/machine_learning/neural_network_train.py]: Dropped the selu / lecun stuff. We're now just running a basic relu neural network. Most people are more familiar with this style of network than they are with selu-based nets

File: .gitattributes [hunk 4]
```diff
@@ -1 +1,2 @@
 *.ipynb filter=lfs diff=lfs merge=lfs -text
+examples/machine_learning/Data.csv filter=lfs diff=lfs merge=lfs -text
```
Comment (timurbazhirov) [inline .gitattributes:2]: let's add a global option for *.csv as well?

File: other/assets/ads_model_sci_adv.csv [hunk 5]
```diff
@@ -0,0 +1,163 @@
+Morphology,N_Atoms,Element,CN,Adsorbate,BoundMetal,PBE_BE_eV,CE_Local_eV,ChemPot_eV,MADS_eV
```
Comment (timurbazhirov) [inline other/assets/ads_model_sci_adv.csv:1]: Should be in LFS

File: .github/workflows/check_links.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+name: check_links
+
+on: [push, pull_request]
+
+jobs:
```
Comment (timurbazhirov) [inline .github/workflows/check_links.yml:5]: 👍

**Pull request merged**